flink-python/src/main/java/org/apache/flink/streaming/api/operators/python/AbstractPythonFunctionOperator.java

```diff
@@ -344,7 +345,7 @@ protected PythonEnvironmentManager createPythonEnvironmentManager() throws IOExc
 			return new ProcessPythonEnvironmentManager(
 				dependencyInfo,
 				getContainingTask().getEnvironment().getTaskManagerInfo().getTmpDirectories(),
-				System.getenv());
+				new HashMap<>(System.getenv()));
```

Comment:
Why need this change?

Reply:
Because System.getenv() returns a Collections.unmodifiableMap that we can not set any new env variables, therefore we will put it into a new HashMap.